examplebroker/broker.go

```diff
@@ -140,6 +140,11 @@ func (b *Broker) NewSession(ctx context.Context, username, lang, mode string) (s
 		return "", "", fmt.Errorf("user %q does not exist", username)
 	}
 
+	if info.sessionMode == "passwd" {
+		info.neededAuthSteps++
+		info.pwdChange = mustReset
+	}
+
```

Comment:
This is a very timid implementation on the ExampleBroker side, but I think that if we store the passwords here and then run second calls to see if it was changed or not will be us testing the ExampleBroker rather than PAM itself, that's why I didn't add anything more. Let me know if you find it necessary!

Reply:
well, the thing is then we don't know if the password token was properly transmitted to the broker in the integration tests then, no? So I was supposing the integration test would have a second password only authentication call with the new password.

Reply:
I could swear we were matching the newpassword to be `== newpassword`, but turns out we don't :joy: I'll update it to do some matching to ensure the data was passed correctly.